ignite/pkg/protoanalysis/package.go

```diff
@@ -1,58 +1,61 @@
 package protoanalysis
 
 import (
+	"errors"
+	"regexp"
 	"strings"
 
-	"github.com/pkg/errors"
+	"golang.org/x/mod/semver"
 )
 
-type Packages []Package
+type (
+	// Packages represents slice of Package.
+	Packages []Package
 
-func (p Packages) Files() Files {
-	var files []File
-	for _, pkg := range p {
-		files = append(files, pkg.Files...)
-	}
-	return files
-}
-
-// Package represents a proto pkg.
-type Package struct {
-	// Name of the proto pkg.
-	Name string
+	PkgName string
 
-	// Path of the package in the fs.
-	Path string
+	// Package represents a proto pkg.
+	Package struct {
+		// Name of the proto pkg.
+		Name string
 
-	// Files is a list of .proto files in the package.
-	Files Files
+		// Path of the package in the fs.
+		Path string
 
-	// GoImportName is the go package name of proto package.
-	GoImportName string
+		// Files is a list of .proto files in the package.
+		Files Files
 
-	// Messages is a list of proto messages defined in the package.
-	Messages []Message
+		// GoImportName is the go package name of proto package.
+		GoImportName string
 
-	// Services is a list of RPC services.
-	Services []Service
-}
+		// Messages is a list of proto messages defined in the package.
+		Messages []Message
 
-type Files []File
+		// Services is a list of RPC services.
+		Services []Service
+	}
+)
 
-type File struct {
-	// Path of the file.
-	Path string
+var regexBetaVersion = regexp.MustCompile("^(v)([0-9]+)(beta|alpha)([0-9]+)")
```

Comment:
Regexp could also be written as:

```suggestion
var regexBetaVersion = regexp.MustCompile("^v[0-9]+(beta|alpha)[0-9]+")
```